bitmap-weasels/lib/read-write.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+const readWrite = module.exports = {};
+
+const tranf = require('./transform');
+const Bitmap = require('./bitmap');
+const fs = require('fs');
+
+readWrite.bmp;
+readWrite.nameKeyThing;
+
+readWrite.read = () => {
```

Comment:
Rather than the nested calls to accomplish this, I would recommend passing an argument to the `.read` method for the transform. That should then kick off just one invocation of `.read`, and use whatever transform specified in the call. 

Back in index.js, you can call `.read('transform')` a few times to do each of your three transforms.